generator/docs/ProfileValidation.md

```diff
@@ -0,0 +1,175 @@
+# Profile Validation
+
+The profile validator analyzes a profile to check if it is valid. A valid profile is one that can generate data. When an invalid profile is observed, the process will terminate and notify the user of the validation errors in the profile. 
+
+The validation checks currently performed include:
+
+1. All data values are excluded from the possible range
+
+    Example: A is in set [3] and A is not in set [3] - there are no values that A can be.
```

Comment:
Hi @sl-slaing, The existing implementation generates no data. However this might be incorrect depending on whether by null we mean empty set. 

https://www.wolframalpha.com/input/?i=is+the+set+(%7B3%7D+-+%7B3%7D)+%3D+emptyset

See SetRestrictions.java / merge()... 

     if (newWhitelist != null && newWhitelist.size() == 0) {
            return new MergeResult<>();
        }

example profile to test with: 

{
  "schemaVersion" : "v3",
  "fields" : [
  {
  "name": "report_type"},
  {
  "name": "age"}
  
  ],
  "rules" : [ 
{
      "field": "report_type",
      "is": "inSet",
      "values": [
        "Derivative Transaction"
      ]
    },
	{"not":
{
      "field": "report_type",
      "is": "inSet",
      "values": [
        "Derivative Transaction"
      ]
    }}
	
     ]
  } ]
}

Reply:
added to #323